www/js/survey/enketo/AddNoteButton.tsx

```diff
@@ -80,16 +81,8 @@ const AddNoteButton = ({ timelineEntry, notesConfig, storeKey }) => {
     return EnketoSurveyLaunch
       .launch($rootScope, surveyName, { timelineEntry, prefillFields, dataKey: storeKey })
       .then(result => {
-        if (!result) {
-          return;
-        }
-        const addition = {
-          data: result,
-          write_ts: Date.now(),
-          key: storeKey,
-        };
-        // adding the addition for display is handled in infinite_scroll_list.js
-        $rootScope.$broadcast('enketo.noteAddition', addition);
```

Comment:
I want to avoid using `rootScope` and `broadcast`. I also want to avoid mutating the trip/place object.

`repopulateTimelineEntry` updates the trip/place in an immutable, React-friendly way.